lib/test/Test/Pos/ConfigurationSpec.hs

```diff
@@ -0,0 +1,49 @@
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+-- | This module tests configuration.
+
+module Test.Pos.ConfigurationSpec
+       ( spec
+       ) where
+
+import           Universum
+
+import           System.Wlog (HasLoggerName (..), LoggerName (..))
+
+import           Mockable.Production (Production (..))
+import           Pos.Core.Slotting (getCurrentTimestamp)
+import           Pos.Crypto (ProtocolMagic (..), RequiresNetworkMagic (..))
+import           Pos.Infra.Ntp.Configuration (NtpConfiguration)
+import           Pos.Launcher.Configuration (ConfigurationOptions (..), withConfigurations)
+
+import           Test.Hspec (Spec, describe, it, runIO, shouldBe)
+
+instance HasLoggerName IO where
+    askLoggerName = pure $ LoggerName "ConfigurationSpec"
+    modifyLoggerName _ x = x
+
+configFilePath :: FilePath
+configFilePath = "configuration.yaml"
+
+checkYamlSection :: Text -> Spec
+checkYamlSection key = describe ("key: " ++ show key) $ do
+    startTime <- runIO (runProduction getCurrentTimestamp)
+    let opts = ConfigurationOptions configFilePath key (Just startTime) Nothing
+    rnm <- runIO (withConfigurations Nothing opts getRNM)
+    it "should be NMMustBeNothing" $
+        rnm `shouldBe` NMMustBeNothing
+
+getRNM :: NtpConfiguration -> ProtocolMagic -> IO RequiresNetworkMagic
+getRNM _ pm = pure (getRequiresNetworkMagic pm)
+
+spec :: Spec
+spec = mapM_ checkYamlSection [
+    "mainnet_full" -- mainnet core/relay nodes and exchange wallets
```

Comment:
Missing commas - I'll amend the commit and force-push.

Reply:
thanks, I make that mistake all the time